gumbo_client/client_v2.py

```diff
@@ -0,0 +1,46 @@
+import os
+from google.oauth2 import service_account
+from google.auth.transport.requests import AuthorizedSession
+
+import pandas as pd
+
+base_url = 'https://client-api-dot-depmap-gumbo.uc.r.appspot.com'
+gumbo_configs_dir = os.path.expanduser('~') + "/.config/gumbo/"
+client_id_filename = 'iap_client_id.txt'
+credentials_filename = 'client-iap-auth-sa.json'
+
+
+class Client:
+    def __init__(self):
+        # Read secrets from file 
+        with open(gumbo_configs_dir + client_id_filename, 'r') as file:
+            client_id = file.read().rstrip()
+        creds = service_account.IDTokenCredentials.from_service_account_file(
+            gumbo_configs_dir + credentials_filename,
+            target_audience=client_id)
+        # Get an authed session token 
+        self.authed_session = AuthorizedSession(creds)
+
+    def get(self, table_name: str) -> pd.DataFrame:
```

Comment:
I'm wondering how we get the username in the service. I see you're fetching the `iap_client_id.txt` and storing it as part of the install instructions. So, it's really at that point that we have access to information about the "current user".

What would you think about adding a step like:

```
gcloud config get account > ~/.config/gumbo/username
```

and then in the client we could read that file to find out what were the credentials used to install the client.